src/Microsoft.Health.Fhir.Liquid.Converter/DotLiquids/TemplateLocalFileSystem.cs

```diff
@@ -38,46 +38,45 @@ public string ReadTemplateFile(Context context, string templateName)
 
         public Template GetTemplate(Context context, string templateName)
         {
-            var templatePath = (string)context[templateName];
-            if (templatePath == null)
+            var templateKey = (string)context[templateName];
+            if (templateKey == null)
             {
                 throw new RenderException(FhirConverterErrorCode.TemplateNotFound, string.Format(Resources.TemplateNotFound, templateName));
             }
 
-            return GetTemplate(templatePath) ?? throw new RenderException(FhirConverterErrorCode.TemplateNotFound, string.Format(Resources.TemplateNotFound, templatePath));
+            return GetTemplate(templateKey) ?? throw new RenderException(FhirConverterErrorCode.TemplateNotFound, string.Format(Resources.TemplateNotFound, templateKey));
         }
 
-        public Template GetTemplate(string templateName)
+        public Template GetTemplate(string templateKey)
         {
-            if (string.IsNullOrEmpty(templateName))
+            if (string.IsNullOrEmpty(templateKey))
             {
                 return null;
             }
 
             // Get template from cache first
-            if (_templateCache.ContainsKey(templateName))
+            if (_templateCache.ContainsKey(templateKey))
             {
-                return _templateCache[templateName];
+                return _templateCache[templateKey];
             }
 
             // If not cached, search local file system
-            var templateContent = ReadTemplateFile(templateName);
-            var template = IsCodeMappingTemplate(templateName) ? TemplateUtility.ParseCodeMapping(templateContent) : TemplateUtility.ParseTemplate(templateName, templateContent);
-            var key = IsCodeMappingTemplate(templateName) ? $"{templateName}/{templateName}" : templateName;
+            var templateContent = ReadTemplateFile(templateKey);
 
+            Template template = TemplateUtility.ParseTemplate(templateKey, templateContent);
             if (template != null)
```

Comment:
Maybe even if the `template` is `null` it still needs to be cached? Since the `null` value is still need be parsed and be treated as valid return value right? So caching it makes sense.

Reply:
Updated